_source/_posts/2022-06-15-terraform-eks-microservices.md

```diff
@@ -0,0 +1,622 @@
+      "managed_nodegroup_launch_template_latest_version" = []
+      "managed_nodegroup_status" = [
+        "ACTIVE",
+      ]
+    }
+  },
+])
+region = "eu-west-1"
+vpc_cidr = "10.0.0.0/16"
+vpc_private_subnet_cidr = [
+  "10.0.10.0/24",
+  "10.0.11.0/24",
+  "10.0.12.0/24",
+]
+vpc_public_subnet_cidr = [
+  "10.0.0.0/24",
+  "10.0.1.0/24",
+  "10.0.2.0/24",
+]
+```
+
+You should see the cluster details if you run `kdash` or `kubectl get nodes` commands.
+
+{% img blog/jhipster-k8s-eks-terraform/eks-cluster.png alt:"EKS cluster in KDash" width:"900" %}{: .center-image }
+
+## Set up OIDC authentication using Okta
+
+You can proceed to deploy the sample application. You could skip this step if you used a sample that does not use Okta or OIDC for authentication.
+
+First, navigate to the **store** application folder.
+
+{% include setup/cli.md type="jhipster" %}
+
+> **Note**: Make sure to add the newly created `.okta.env` file to your `.gitignore` file so that you don't accidentally expose your credentials to the public.
+
+Update `kubernetes/registry-k8s/application-configmap.yml` with the OIDC configuration from the `.okta.env` file. The Spring Cloud Config server reads from this file and shares the values with the gateway and microservices.
+
+```yaml
+data:
+  application.yml: |-
+    ...
+    spring:
+      security:
+        oauth2:
+          client:
+            provider:
+              oidc:
+                issuer-uri: https://<your-okta-domain>/oauth2/default
+            registration:
+              oidc:
+                client-id: <client-id>
+                client-secret: <client-secret>
+```
+
+Next, configure the JHipster Registry to use OIDC for authentication. Modify `kubernetes/registry-k8s/jhipster-registry.yml` to enable the `oauth2` profile, which is pre-configured in the JHipster Registry application.
+
+```yaml
+- name: SPRING_PROFILES_ACTIVE
+  value: prod,k8s,oauth2
+```
+
+The application is now ready.
+
+### Secure secrets
+
+If you have noticed, you are setting secrets in plain text on the `application-configmap.yml` file, which is not ideal and is not a best practice for security. For the specific JHipster application, you can use the encrypt functionality provided by the JHipster Registry to encrypt the secrets. See [Encrypt Your Secrets with Spring Cloud Config](/blog/2021/06/01/kubernetes-spring-boot-jhipster#encrypt-your-secrets-with-spring-cloud-config) to learn how to do this. But that would also rely on a base64 encoded encryption key added as a Kubernetes Secret, which still can be decoded. The best way to do this would be using [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/), an external service like [HashiCorp Vault](https://www.hashicorp.com/products/vault), or with [Sealed Secrets](https://github.com/bitnami-labs/sealed-secrets). To learn more about these methods see [Encrypt Your Kubernetes Secrets](/blog/2021/06/01/kubernetes-spring-boot-jhipster#encrypt-your-kubernetes-secrets).
```

Comment:
```suggestion
If you have noticed, you are setting secrets in plain text on the `application-configmap.yml` file, which is not ideal and is not a best practice for security. For the specific JHipster application, you can use the encrypt functionality provided by the JHipster Registry to encrypt the secrets. See [Encrypt Your Secrets with Spring Cloud Config](/blog/2021/06/01/kubernetes-spring-boot-jhipster#encrypt-your-secrets-with-spring-cloud-config) to learn how to do this. But that would also rely on a base64 encoded encryption key added as a Kubernetes Secret, which still can be decoded. The best way to do this would be to use [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/), an external service like [HashiCorp Vault](https://www.hashicorp.com/products/vault), or [Sealed Secrets](https://github.com/bitnami-labs/sealed-secrets). To learn more about these methods see [Encrypt Your Kubernetes Secrets](/blog/2021/06/01/kubernetes-spring-boot-jhipster#encrypt-your-kubernetes-secrets).
```